src/proc.cpp

```diff
@@ -702,20 +703,20 @@ static int select_try(job_t *j) {
     }
 
     if (maxfd >= 0) {
-        int retval;
-        struct timeval tv;
+        struct timeval timeout;
 
-        tv.tv_sec = 0;
-        tv.tv_usec = 10000;
+        timeout.tv_sec = 0;
+        timeout.tv_usec = 10000;
```

Comment:
Btw: Why is this 10ms? Even millliseconds is really quite long in computer time, so wouldn't e.g. 1ms suffice?

Reply:
This used to be 10 _seconds_ from the initial revision up to 6b8e7b16f6d4e11e168e3ce2effe2d8f0a53b184.

https://sourceforge.net/p/fish/mailman/message/23293756/ is the mail that included the patch, and the justification for 10ms seems to be "shorter than 5s, still long enough that it's not busy-waiting". But 10ms still seems to be _really_ long.

Reply:
Good job digging that up. 10ms is long, but 1ms timeouts have a noticeable CPU cost. I would rather just leave this at 10ms for now until we do the concurrency changes we were discussing, at which point the timeout can become `-1` and fish can properly react to both IO and child process exits without polling.